internal/controller/nonadminrestore_controller.go

```diff
@@ -333,6 +300,27 @@ func (r *NonAdminRestoreReconciler) createVeleroRestore(ctx context.Context, log
 	}
 
 	if veleroRestore == nil {
+		if meta.IsStatusConditionTrue(nar.Status.Conditions, string(nacv1alpha1.NonAdminConditionQueued)) {
+			err = errors.New("NonAdminRestore is in final stage and its related Velero Restore was deleted")
```

Comment:
Same question here - aren’t we introducing inconsistent behavior between Non-Admin Backup and Velero Backup with extra logic that may not be needed?

Reply:
from meeting today, we will only have status message saying related object was deleted